src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs

```diff
@@ -1618,17 +1610,11 @@ private void DoCreateRunspace(string initialCommand, bool skipProfiles, bool sta
 
         private void OpenConsoleRunspace(Runspace runspace, bool staMode)
         {
-#if STAMODE
-            // staMode will have following values:
-            // On FullPS: 'true'/'false' = default('true'=STA) + possibility of overload through cmdline parameter '-mta'
-            // On NanoPS: always 'false' = default('false'=MTA) + NO possibility of overload through cmdline parameter '-mta'
-            // ThreadOptions should match on FullPS and NanoPS for corresponding ApartmentStates.
-            if (staMode)
+            if (staMode && !Platform.IsNanoServer)
```

Comment:
Should we check for `Platform.IsIoT` here?
And, this check means on macOS and Linux, if `staMode` is true, we will go through this code path. Is it guaranteed that on macOS and Linux, `staMode` is never true?

Reply:
The current code guarantees `staMode` is never true on non-Windows.  Will change to use `Platform.IsWindowsDesktop`